File: src/lessons/welcome/bdr/BDR2Entity.c
#include "RemoteBuggle.h"

char getIndication() {
	if (isOverMessage()) {
		return readMessage()[0];
	} else {
		return ' ';
	}
}

int main(){
	#line 1 "BDR2"
	/* BEGIN SOLUTION */
	int continu=1;
	char read;
	while(continu){

		read = getIndication();
		switch (read) {
			case 'R': right(); forward(1); break;
			case 'L': left();  forward(1); break;
			case 'I': back();  forward(1); break;

			case 'A': forward(1); break;
			case 'B': forward(2); break;
			case 'C': forward(3); break;
			case 'D': forward(4); break;
			case 'E': forward(5); break;
			case 'F': forward(6); break;

			case 'Z': backward(1); break;
			case 'Y': backward(2); break;
			case 'X': backward(3); break;
			case 'W': backward(4); break;
			case 'V': backward(5); break;
			case 'U': backward(6); break;
			default : continu=0;
		}

	}
	/* END SOLUTION */
	return 0;
}

File: src/lessons/welcome/bdr/BDR2Entity.scala
package lessons.welcome.bdr;

import java.util.Stack;

import plm.universe.bugglequest.SimpleBuggle;

class BDR2Entity extends SimpleBuggle {
	override def run() { 
		/* BEGIN SOLUTION */
		var moreMusic = true;

		while (moreMusic) {
			readMessage() match {
			  case "R" => right(); stepForward();
			  case "L" => left();  stepForward();
			  case "I" => back();  stepForward();

			  case "A" => forward(1);
			  case "B" => forward(2);
			  case "C" => forward(3);
			  case "D" => forward(4);
			  case "E" => forward(5);
			  case "F" => forward(6);

			  case "Z" => backward(1);
			  case "Y" => backward(2);
			  case "X" => backward(3);
			  case "W" => backward(4);
			  case "V" => backward(5);
			  case "U" => backward(6);
			  
			  case _ =>  moreMusic = false
			}
		}
		/* END SOLUTION */
	}
}
